File: Data Structures & Algorithms/binary-tree-diameter/submission-1.go
/**
 * Definition for a binary tree node.
 * type TreeNode struct {
 *     Val int
 *     Left *TreeNode
 *     Right *TreeNode
 * }
 */

func diameterOfBinaryTree(root *TreeNode) int {
    ans := 0
    var h func(node *TreeNode) int 
    h = func(node *TreeNode) int {
        if node == nil {
            return 0
        }

        l := h(node.Left)
        r := h(node.Right)
        ans = max(ans, l + r)

        return max(l, r) + 1
    }
    h(root)
    return ans
}

File: Data Structures & Algorithms/kth-smallest-integer-in-bst/submission-1.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    def kthSmallest(self, root: Optional[TreeNode], k: int) -> int:
        ans = None

        def m(node):
            nonlocal k, ans

            if not node or ans is not None:
                return

            m(node.left)

            k -= 1
            if k == 0:
                ans = node.val
                return
            
            m(node.right)

        m(root)
        return ans